src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -140,39 +117,32 @@ public static Logger getLogger() {
 
   public static void gameOverClearCache() {
     // Is static, set to null so that we don't keep the data around after a game is exited
-    s_battleCalculator.setGameData(null);
+    ProBattleUtils.setData(null);
     ProLogUI.clearCachedInstances();
   }
 
-  public IOddsCalculator getCalc() {
-    return s_battleCalculator;
-  }
-
-  @Override
-  public final GameData getGameData() {
-    if (data != null) {
-      return data;
-    } else {
-      return super.getGameData();
-    }
-  }
-
   @Override
   public void stopGame() {
     super.stopGame(); // absolutely MUST call super.stopGame() first
-    s_battleCalculator.cancel(); // cancel any current calcing
+    ProBattleUtils.cancel(); // cancel any current calcing
```

Comment:
True, but I do want to point out this type of problem. Comments and code are a form of coupling, one more extra point of maintenance. Every line of code and every line of comment both have their cost, they need to justify their existence or else be removed.

I suppose an analogy is drafts in a building. A single window draft wont' hurt, but if you have thousands of windows and they are all drafty, the problem gets to be bad. In this case the problem com  when you want to make changes. Given the coupling, and the already near zero value of the comments, chances are much better the behavior of the cancel method will be altered and then this comment wont' be updated, as comments are generally not updated, and then it'll just be an elaborate lie rather than something useful.